Improvement/896 add pillar for bootstrap services ip

ramanduh opened the pull request:
The goal of this PR is to create a pillar named `endpoints` that will contain the IP of pods, running on the bootstrap.
To perform that, we create a service that map to these pods, and we get the pods IP from the endpoints object created with the service.

To test: run the bootstrap, an then query the pillar `metalk8s:endpoints`
```
# crictl exec -ti $(crictl ps | awk '/salt-master/{print $1}') salt "bootstrap" pillar.get metalk8s:endpoints saltenv=metalk8s-2.0
bootstrap:
    ----------
    package-repositories:
        ----------
        hostname:
            None
        ip:
            172.21.254.3
        node_name:
            bootstrap
        ports:
            ----------
            http:
                8080
    registry:
        ----------
        hostname:
            None
        ip:
            172.21.254.3
        node_name:
            bootstrap
        ports:
            ----------
            registry:
                5000
    salt-master:
        ----------
        hostname:
            None
        ip:
            172.21.254.3
        node_name:
            bootstrap
        ports:
            ----------
            api:
                4507
            publisher:
                4505
            requestserver:
                4506

```

Commit:
Add a function to show and endpoint in kubernetes module

Relates #896

Commit:
Add bootstrap service IP in pillar

Relates #896

Commit:
Add an kubernetes headless service for salt-master

The goal of this service is to track the salt-master pod IP from the service
endpoint

Closes #896

bert-e commented:
# Hello ramanduh,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-add-pillar-for-bootstrap-services-ip` into
`development/2.0`
* [w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.1` 
* [w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

bert-e commented:
# History mismatch



Merge commit #229e1614f57a6396879a2a8e4d71b1368d930036 on the integration branch
`w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip` is merging a branch which is neither the current
branch `improvement/896-add-pillar-for-bootstrap-services-ip` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/896-add-pillar-for-bootstrap-services-ip` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

Commit:
Add a kubernetes headless service for registry

Closes #896

Commit:
Add a kubernetes headless service package repository

The goal of this service is to track the package repository pod IP from the
service endpoint

Closes #896

NicolasT commented on salt/_pillar/metalk8s.py:
```diff
@@ -49,12 +51,42 @@ def _load_ca(config_data):
     }
 
 
+def _get_endpoints(config="/etc/kubernetes/admin.conf",
```
I'd like to keep pillars-that-dont-need-Kubernetes-available and pillars-that-need-Kubernetes-available separate (in separate modules), so this shouldn't go in here.

ramanduh replied:
Do you mean move it in `metalk8s_nodes.py` or create another module? BTW I just see that in `metalk8s_nodes.py` we use directly the `python-kubernetes` python client. In this PR context, is it preferable to use  `python-kubernetes` client directly or do we want to use salt client and then maintain `salt/_modules/kubernetesmod.py`?

NicolasT replied:
I'd say, add another pillar module.

You can keep using the `__salt__` procedures, that's fine. The other one uses `python-kubernetes` directly since it's based on some work I did a long time ago. So is the roster module, actually.

I guess over time we should harmonize those.

NicolasT commented on salt/_pillar/metalk8s.py:
```diff
@@ -49,12 +51,42 @@ def _load_ca(config_data):
     }
 
 
+def _get_endpoints(config="/etc/kubernetes/admin.conf",
```
Don't use defaults like this, they're arbitrary. Values should be passed in from configuration.

ramanduh replied:
will fix

NicolasT commented on salt/_pillar/metalk8s.py:
```diff
@@ -49,12 +51,42 @@ def _load_ca(config_data):
 
+def _get_endpoints(config="/etc/kubernetes/admin.conf",
+                   context="kubernetes-admin@kubernetes",
+                   namespace="kube-system"):
```
Making `namespace` an input argument, but `service` not, isn't very useful.

No need to make any of this configurable, IMHO. We know what this `ext_pillar` is supposed to provide. If more, then code changes are required.

ramanduh replied:
will fix

NicolasT commented on salt/_pillar/metalk8s.py:
```diff
@@ -49,12 +51,42 @@ def _load_ca(config_data):
+            or not os.path.exists(config):
+        return endpoints
+
+    for service in ('salt-master', 'registry', 'package-repositories'):
```
These strings form a list or a set, not a tuple.

ramanduh replied:
will fix

NicolasT commented on salt/_pillar/metalk8s.py:
```diff
@@ -49,12 +51,42 @@ def _load_ca(config_data):
+                kubeconfig=config,
+                context=context
+            )
+            endpoint_ip = endpoint["subsets"][0]["addresses"][0]['ip']
```
Why throw away port information?

ramanduh replied:
No specific reason, I just thought that we'll only use ip in our use case, but it makes sense to add port info as well. BTW, we'll have something like this in the `endpoint["subsets"]` dict:
```
>>> pprint(coreV1.read_namespaced_endpoints('salt-master', 'kube-system').to_dict()['subsets'])
[{'addresses': [{'hostname': None,
                 'ip': '172.21.254.3',
                 'node_name': 'bootstrap',
                 'target_ref': {'api_version': None,
                                'field_path': None,
                                'kind': 'Pod',
                                'name': 'salt-master-bootstrap',
                                'namespace': 'kube-system',
                                'resource_version': '577',
                                'uid': 'f857e39e-5d26-11e9-8308-5254008481d5'}}],
  'not_ready_addresses': None,
  'ports': [{'name': 'requestserver', 'port': 4506, 'protocol': 'TCP'},
            {'name': 'publisher', 'port': 4505, 'protocol': 'TCP'},
            {'name': 'api', 'port': 4507, 'protocol': 'TCP'}]}]
```
Which form do you recommend to return the pillar? Return the dict as it is or make some processing to simplify it (in the pillar I only take the first adress in the subset, but I wonder if it is a good idea)

NicolasT replied:
Taking 'the first one' is fine, for now there will be only one anyway, since the bootstrap node is not HA right now anyway.

I think what you may want to return is something like the following:

```json
[
  {
    "node_name": "bootstrap",
    "hostname": null,
    "ip": "172.21.254.3",
    "ports": {
      "requestserver": 4506,
      "publisher": 4505,
      "api": 4507
    }
  }
]
```

Then, in the *ext_pillar*, we set the pillar value to the 'first' element of what this procedure returns, and then consumer of that pillar value can use something like

```jinja
foo: {{ salt_master.ip }}:{{ salt_master.ports.publisher }}
```

or something along those lines.

NicolasT commented on salt/metalk8s/orchestrate/bootstrap_with_master.sls:
```diff
@@ -27,6 +27,17 @@ Bootstrap control plane:
+    - require:
+      - salt: Bootstrap control plane
+    - pillar:
+        registry_ip: {{ pillar.get('registry_ip') }}
```
Is this used?

ramanduh replied:
Not used, will remove it

NicolasT reviewed (changes requested):
Couple of requests.

gdemonet commented on buildchain/buildchain/salt_tree.py:
```diff
@@ -113,6 +113,7 @@ def task__deploy_salt_tree() -> Iterator[types.TaskDict]:
     Path('salt/metalk8s/bootstrap/preflight.sls'),
     Path('salt/metalk8s/bootstrap/registry.sls'),
     Path('salt/metalk8s/bootstrap/salt_master.sls'),
+    Path('salt/metalk8s/bootstrap/services.sls'),
```
So, I was wondering, do we need such a formula? If we follow the structure defined in https://github.com/scality/metalk8s/pull/967, I guess we could have a `deployed.sls` in all "services", i.e. `metalk8s.registry`, `metalk8s.repo` and `metalk8s.salt.master`. All these will be called as soon as we have an API server ready.

ramanduh replied:
@TeddyAndrieux any thought on this? Is there anything I can change in my PR to facilitate https://github.com/scality/metalk8s/pull/967 ? FYI: for now I can move the service object deployment in `salt/metalk8s/salt/master/deployed.sls` because when the state deployed.sls is applied, the API server is not up yet. Does it make sense for you if this change is managed in https://github.com/scality/metalk8s/pull/967 ?

TeddyAndrieux replied:
As for the moment we already have `deployed.sls` for salt master and repo (which is not real "deployed" as we want to use in our salt states)
I think you should keep the naming `service_deployed.sls` for the moment. I will rename them in my PR

bert-e commented:
# History mismatch



Merge commit #88560990212e7d2aebd8c6eff5b83419c226083c on the integration branch
`w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip` is merging a branch which is neither the current
branch `improvement/896-add-pillar-for-bootstrap-services-ip` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/896-add-pillar-for-bootstrap-services-ip` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

NicolasT commented on salt/_pillar/metalk8s_endpoints.py:
```diff
@@ -0,0 +1,56 @@
+        endpoint = __salt__['kubernetes.show_endpoint'](
+            name=service,
+            namespace='kube-system',
+            kubeconfig='/etc/kubernetes/admin.conf',
```
These should ideally come from the `ext_pillar` configuration, not hard-coded.

ramanduh replied:
I'll move namespace and kubeconfig to ext_pillar configuration

ramanduh commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-add-pillar-for-bootstrap-services-ip` into
`development/2.0`
* [w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.1` 
* [w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

Commit:
Make context arguement not mandatory

If context is not specified, let python-kubernetes manage it. The
default behaviour is to use the current_context from config file.

Relates #896

bert-e commented:
# History mismatch



Merge commit #4bf596834aad7c8c4dccc9a58955b07776bcb19b on the integration branch
`w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip` is merging a branch which is neither the current
branch `improvement/896-add-pillar-for-bootstrap-services-ip` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/896-add-pillar-for-bootstrap-services-ip` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ramanduh commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-add-pillar-for-bootstrap-services-ip` into
`development/2.0`
* [w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.1` 
* [w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

NicolasT commented on salt/_pillar/metalk8s_endpoints.py:
```diff
@@ -0,0 +1,57 @@
+
+log = logging.getLogger(__name__)
+
+__virtualenv__ = 'metalk8s_endpoints'
```
It's `__virtualname__`, not `__virtualenv__`

ramanduh replied:
oups, will fix

bert-e commented:
# History mismatch



Merge commit #cd51a9d4014ab033b7debe1a18011be86ea6f80a on the integration branch
`w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip` is merging a branch which is neither the current
branch `improvement/896-add-pillar-for-bootstrap-services-ip` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/896-add-pillar-for-bootstrap-services-ip` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ramanduh commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-add-pillar-for-bootstrap-services-ip` into
`development/2.0`
* [w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.1` 
* [w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

NicolasT commented on salt/_pillar/metalk8s_endpoints.py:
```diff
@@ -0,0 +1,57 @@
+
+
+def __virtual__():
+    return __virtualname__
```
Strictly speaking this should indicate failure if `kubernetes.show_endpoint` can't be found in `__salt__`, but fine.

NicolasT commented on salt/_pillar/metalk8s_endpoints.py:
```diff
@@ -0,0 +1,57 @@
+        }
+        res['ports'] = ports
+    except Exception as exc:  # pylint: disable=broad-except
+        log.error(
```
Could use `log.exception` instead.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

bert-e commented:
# History mismatch



Merge commit #86c63b4e414a4061fb9527eef0a17de85dc17b35 on the integration branch
`w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip` is merging a branch which is neither the current
branch `improvement/896-add-pillar-for-bootstrap-services-ip` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/896-add-pillar-for-bootstrap-services-ip` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ramanduh commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-add-pillar-for-bootstrap-services-ip` into
`development/2.0`
* [w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.1` 
* [w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

ramanduh commented:
@bert-e help

bert-e commented:
# Help page



The following options and commands are available at this time.

## Options ##


name   | description  | privileged   | authored
------ | ------------ | ------------ |----------
:arrow_right: **approve** | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
:arrow_right: **unanimity** | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
:arrow_right: **bypass_leader_approval** | Bypass the pull request leaders' approval |  :star:  | 
:arrow_right: **bypass_peer_approval** | Bypass the pull request peers' approval |  :star:  | 
:arrow_right: **bypass_jira_check** | Bypass the Jira issue check |  :star:  | 
:arrow_right: **after_pull_request** | Wait for the given pull request id to be merged before continuing with the current one. |  | 
:arrow_right: **no_octopus** | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
:arrow_right: **wait** | Instruct Bert-E not to run until further notice. |  | 
:arrow_right: **bypass_commit_size** | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
:arrow_right: **create_pull_requests** | Allow the creation of integration pull requests. |  | 
:arrow_right: **bypass_incompatible_branch** | Bypass the check on the source branch prefix |  :star:  | 
:arrow_right: **bypass_build_status** | Bypass the build and test status |  :star:  | 
:arrow_right: **bypass_author_approval** | Bypass the pull request author's approval |  :star:  | 



## Commands ##


name   | description  | privileged
------ | ------------ | ------------
:arrow_right: **status** | Print Bert-E's current status in the pull request ```TBA``` | 
:arrow_right: **reset** | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
:arrow_right: **clear** | Remove all comments from Bert-E from the history ```TBA``` | 
:arrow_right: **force_reset** | Delete integration branches & pull requests, and restart merge process from the beginning. | 
:arrow_right: **retry** | Re-start a fresh build ```TBA``` | 
:arrow_right: **build** | Re-start a fresh build ```TBA``` | 
:arrow_right: **help** | Print Bert-E's manual in the pull request. |

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

ramanduh commented:
@bert-e after_pull_request 1009

bert-e commented:
# Incorrect command syntax



It seems that your command syntax is incorrect. The correct usage is:

```
@bert-e option[=argument]
```

Please **edit** or **delete** the corresponding comment so I can move on.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

ramanduh commented:
@bert-e after_pull_request=1009

bert-e commented:
# Waiting for other pull request(s)



The current pull request is locked by the **after_pull_request** option.

In order for me to merge this pull request, run the following actions first:


:arrow_right: Merge the `OPEN` pull request:

* pull request #1009





Alternatively, delete all the **after_pull_request** comments from this pull request.





*The following options are set:* **after_pull_request**

ramanduh commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **after_pull_request**

bert-e commented:
# Waiting for other pull request(s)



The current pull request is locked by the **after_pull_request** option.

In order for me to merge this pull request, run the following actions first:


:arrow_right: Merge the `OPEN` pull request:

* pull request #1009





Alternatively, delete all the **after_pull_request** comments from this pull request.





*The following options are set:* **after_pull_request**

TeddyAndrieux commented:
@bert-e status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **after_pull_request**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-add-pillar-for-bootstrap-services-ip` into
`development/2.0`
* [w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.1` 
* [w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **after_pull_request**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf










*The following options are set:* **after_pull_request**

bert-e commented:
# History mismatch



Merge commit #714b3db4e479c92a56af2c255d8d70c8f1dcceb4 on the integration branch
`w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip` is merging a branch which is neither the current
branch `improvement/896-add-pillar-for-bootstrap-services-ip` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/896-add-pillar-for-bootstrap-services-ip` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **after_pull_request**

TeddyAndrieux commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **after_pull_request**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-add-pillar-for-bootstrap-services-ip` into
`development/2.0`
* [w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.1` 
* [w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-add-pillar-for-bootstrap-services-ip) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **after_pull_request**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf










*The following options are set:* **after_pull_request**

NicolasT commented:
@bert-e bypass_author_approval

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **after_pull_request, bypass_author_approval**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye ramanduh.
